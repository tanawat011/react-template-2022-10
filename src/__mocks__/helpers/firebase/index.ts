export const deleteCollection = jest.fn()
export const deleteDocument = jest.fn()
export const getDocument = jest.fn()
export const getAllDocuments = jest.fn()
export const setDocument = jest.fn()
export const subscribeCollection = jest.fn()
export const updateDocument = jest.fn()
