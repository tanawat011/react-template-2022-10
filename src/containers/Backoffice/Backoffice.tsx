import { useEffect } from 'react'

import { Outlet, useLocation, useNavigate } from 'react-router-dom'
import tw from 'twin.macro'

import { PATH } from 'Routes'
import { Button } from 'components/Button'
import { TwCol, TwRow } from 'components/Common'

const TwContainer = tw(TwRow)`w-full h-screen bg-slate-200`
const TwSidebar = tw(TwCol)`min-w-[280px] bg-slate-300`
const TwSidebarHeader = tw(TwRow)`min-h-[64px] bg-slate-400`
const TwSidebarMenu = tw(TwCol)`h-full bg-slate-500`
const TwFeature = tw(TwCol)`w-full overflow-auto bg-slate-600`
const TwFeatureHeader = tw(TwRow)`justify-between min-h-[64px] bg-slate-700`
const TwFeatureContainer = tw(TwRow)`h-full overflow-x-auto bg-slate-800`

export const BackofficeContainer: React.FC = () => {
  const location = useLocation()
  const navigate = useNavigate()

  const { AUTH, BACKOFFICE, ERROR } = PATH
  const { ROOT: GAME, HUES_AND_CUES } = BACKOFFICE.GAME

  const homePath = `../${BACKOFFICE.HOME}`

  useEffect(() => {
    handleRedirectToHome()
  }, [])

  const handleRedirectToHome = () => {
    const defaultPaths = ['', '/']

    if (defaultPaths.includes(location.pathname)) {
      navigate(homePath)
    }
  }

  return (
    <TwContainer>
      <TwSidebar>
        <TwSidebarHeader>Sidebar Header</TwSidebarHeader>

        <TwSidebarMenu>
          <Button link to={homePath}>
            Home
          </Button>
          <Button link to={`../${BACKOFFICE.TODO}`}>
            Todo
          </Button>
          <Button link to={`../${BACKOFFICE.ABOUT}`}>
            About
          </Button>
          <Button link to={`../${GAME}/${HUES_AND_CUES.ROOT}/${HUES_AND_CUES.SET_DISPLAY_NAME}`}>
            Game
          </Button>
          <Button link to={`../${BACKOFFICE.READ_CSV}`}>
            Read CSV
          </Button>

          <h1>Error Stage</h1>

          <Button link to={`/${ERROR[401]}`}>
            Error 401
          </Button>
          <Button link to={`/${ERROR[403]}`}>
            Error 403
          </Button>
          <Button link to={`/${ERROR[404]}`}>
            Error 404
          </Button>
          <Button link to={`/${ERROR[500]}`}>
            Error 500
          </Button>
          <Button link to={`/${ERROR[502]}`}>
            Error 502
          </Button>
          <Button link to={`/${ERROR[503]}`}>
            Error 503
          </Button>
          <Button link to={`/${ERROR[504]}`}>
            Error 504
          </Button>
        </TwSidebarMenu>
      </TwSidebar>

      <TwFeature>
        <TwFeatureHeader>
          <p>Container Header</p>

          <Button link to={`/${AUTH.ROOT}`}>
            Logout
          </Button>
        </TwFeatureHeader>

        <TwFeatureContainer>
          <Outlet />
        </TwFeatureContainer>
      </TwFeature>
    </TwContainer>
  )
}
