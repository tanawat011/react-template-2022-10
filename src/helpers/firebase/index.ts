export * from './deleteCollection'
export * from './deleteDocument'
export * from './getAllDocuments'
export * from './getDocument'
export * from './setDocument'
export * from './subscribeCollection'
export * from './updateDocument'
