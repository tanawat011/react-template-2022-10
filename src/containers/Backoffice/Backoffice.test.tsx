import { renderWithProviders } from 'helpers/test'

import { BackofficeContainer } from './Backoffice'

describe('<BackofficeContainer />', () => {
  test('should render', () => {
    const { container } = renderWithProviders(<BackofficeContainer />)
    expect(container).toBeInTheDocument()
  })

  test('match snapshot', () => {
    const { asFragment } = renderWithProviders(<BackofficeContainer />, {
      router: { route: '/backoffice/home' },
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
