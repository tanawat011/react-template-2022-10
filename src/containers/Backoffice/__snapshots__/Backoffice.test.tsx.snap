// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackofficeContainer /> match snapshot 1`] = `
<DocumentFragment>
  <p>
    BackofficeContainer
  </p>
  <a
    href="/home"
  >
    Home
  </a>
  <a
    href="/todo"
  >
    Todo
  </a>
  <a
    href="/about"
  >
    About
  </a>
  <a
    href="/about2"
  >
    About2
  </a>
  <a
    href="/auth"
  >
    Logout
  </a>
</DocumentFragment>
`;
